lib/collection/src/collection_manager/segments_searcher.rs

```diff
@@ -464,11 +452,19 @@ async fn search_in_segment(
     Ok((result, further_results))
 }
 
-/// None if plain index, Some if hnsw.
-fn config_hnsw_ef_construct(config: SegmentConfig) -> Option<usize> {
+/// Find the maximum segment or vector specific HNSW ef_construct in this config
+///
+/// If the index is `Plain`, `None` is returned.
+fn config_max_hnsw_ef_construct(config: SegmentConfig) -> Option<usize> {
```

Comment:
I reworked this to also consider HNSW configs for specific vectors. It now returns the maximum value.

I'm not entirely sure if this behavior is correct, meaning: I'm not entirely sure this function is intended to work this way. It is not super clear to me based on the two callers.

Someone more knowledgeable should check whether this returning the maximum `ef_construct` value is correct in the places it is used.